heimdall/ext.py

```diff
@@ -0,0 +1,24 @@
+FILE_CONSTRUCTORS = dict()
```

Comment:
Nice design!
I want to refactor all the file related things to an upstream dependency though, and I will be inspired by this ;).